### ISSUE TITLE
Incorrect warnings while RA is still loading

### ISSUE THREAD

**Issue by RalfJung** [rust-lang/rust-analyzer#18190]
When opening [this file](https://github.com/RalfJung/rustc-build-sysroot/blob/6ebd8d6022df6180b5705c71544655757ef23db5/src/lib.rs#L423), RA briefly shows an incorrect warning:

![Image](https://github.com/user-attachments/assets/8e92ac34-8886-4cee-bae8-a2c28ba61d0b)

RA version:
rust-analyzer version: 0.3.2121-standalone [/home/r/.vscode-oss/extensions/rust-lang.rust-analyzer-0.3.2121-linux-x64/server/rust-analyzer]

Settings:
```
    "rust-analyzer.lens.enable": false,
    "rust-analyzer.cachePriming.enable": false,
    "rust-analyzer.cargo.buildScripts.useRustcWrapper": false, // https://github.com/rust-lang/rust-analyzer/issues/14722
    "rust-analyzer.diagnostics.disabled": [
        "unlinked-file",
        "macro-error" // trouble with the xshell `cmd!` macro
    ],
```

**Comment by Veykril** [comment]
I *believe* this is due to us loading the project before loading the standard libraries. We shouldn't compute semantic diagnostics at this loading stage (only parse diagnostics).

**Comment by dev-ardi** [comment]
I also tend to get the "file not included in module tree" error.

**Comment by Veykril** [comment]
That is related but not quite the same. But you are right in that we should surpress those as well until the project is loaded

**Comment by RSS1102** [comment]
> I also tend to get the "file not included in module tree" error.

me too...The program can be run normally, but the prompt is a sad state....😶😶 Even if restart the service

![Image](https://github.com/user-attachments/assets/8b13c783-f130-4ef4-bf8a-097e9b0a6c24)
![Image](https://github.com/user-attachments/assets/8b34313f-c8c0-4928-bc03-3fdb9552b117)

**Comment by ChayimFriedman2** [comment]
@RSS1102 Briefly, or constantly? This issue is about errors/warnings that are shown because the project is not fully loaded and disappear when it finishes loading.

**Comment by RSS1102** [comment]
> [@RSS1102](https://github.com/RSS1102) Briefly, or constantly? This issue is about errors/warnings that are shown because the project is not fully loaded and disappear when it finishes loading.

🥲🥲🥲After many modifications, I think it should be a problem with the file case.
After my investigation:
At first my file was named: `window_Menu.rs`. I changed the file name to `window_menu.rs`, and it prompted `unlink-file`, and asked me to create a `pub mod window_Menu`. (No matter I restarted the rust-analyzer server, restarted vscode, or even restarted the computer, it didn't work.) When I changed the file name to a different `windows_menu.rs` (there is an extra `s`), it worked.

**Comment by ChayimFriedman2** [comment]
@RSS1102  On Windows it is case insensitive, on Linux it is case sensitive (not sure about MacOS). I believe there is a way to rename case in Windows, but I think VSCode does not respect that.

**Comment by RSS1102** [comment]
> [@RSS1102](https://github.com/RSS1102) On Windows it is case insensitive, on Linux it is case sensitive (not sure about MacOS). I believe there is a way to rename case in Windows, but I think VSCode does not respect that.

I think so, maybe there are other methods that need to be further explored.🥹🥹

**Comment by RalfJung** [comment]
@RSS1102 I think you are having a different issue. Please only post in an existing issue when you are sure it is the same. Otherwise, open a new issue, and reference the potential old issues.
